crates/ruff_python_formatter/src/comments/placement.rs

```diff
@@ -158,40 +159,50 @@ fn handle_in_between_except_handlers_or_except_handler_and_else_or_finally_comme
-        // it now depends on the indentation level of the comment if it is a leading comment for e.g.
-        // the following `elif` or indeed a trailing comment of the previous body's last statement.
-        let comment_indentation =
-            whitespace::indentation_at_offset(locator, comment.slice().range().start())
-                .map(str::len)
-                .unwrap_or_default();
+    let Some(AnyNodeRef::ExceptHandlerExceptHandler(except_handler)) = comment.preceding_node() else {
+        return CommentPlacement::Default(comment);
+    };
 
-        if let Some(except_indentation) =
-            whitespace::indentation(locator, except_handler).map(str::len)
+    // it now depends on the indentation level of the comment if it is a leading comment for e.g.
+    // the following `elif` or indeed a trailing comment of the previous body's last statement.
+    let comment_indentation =
+        whitespace::indentation_at_offset(locator, comment.slice().range().start())
+            .map(str::len)
+            .unwrap_or_default();
+
+    let Some(except_indentation) =
+            whitespace::indentation(locator, except_handler).map(str::len) else
         {
-            return if comment_indentation <= except_indentation {
-                // It has equal, or less indent than the `except` handler. It must be a comment
-                // of the following `finally` or `else` block
-                //
-                // ```python
-                // try:
-                //     pass
-                // except Exception:
-                //     print("noop")
-                // # leading
-                // finally:
-                //     pass
-                // ```
-                // Attach it to the `try` statement.
-                CommentPlacement::dangling(comment.enclosing_node(), comment)
-            } else {
-                // Delegate to `handle_trailing_body_comment`
-                CommentPlacement::Default(comment)
-            };
-        }
+            return CommentPlacement::Default(comment);
+        };
+
+    if comment_indentation > except_indentation {
+        // Delegate to `handle_trailing_body_comment`
+        return CommentPlacement::Default(comment);
     }
 
-    CommentPlacement::Default(comment)
+    // It has equal, or less indent than the `except` handler. It must be a comment of a subsequent
+    // except handler or of the following `finally` or `else` block
+    //
+    // ```python
+    // try:
+    //     pass
+    // except Exception:
+    //     print("noop")
+    // # leading
+    // finally:
+    //     pass
+    // ```
+
+    if let Some(following) = comment.following_node() {
+        if let AnyNodeRef::ExceptHandlerExceptHandler(_) = following {
+            // Attach it to the following except handler (which has a node) as leading
+
+            return CommentPlacement::leading(following, comment);
+        }
+    }
```

Comment:
I think this incorrectly attaches the following comment as leading comment rather than trailing `except` comment.

```python
try:
	pass
except Exception:
	print("no-op")
	# trailing
finally:
	pass
```

I wonder if the correct fix is to change the condition on line 179 to delegate to `comment_indentation != except_indentation`.

Reply:
for your example above `--print-comments` prints

```
{
    Node {
        kind: StmtExpr,
        range: 36..50,
        source: `print("no-op")`,
    }: {
        "leading": [],
        "dangling": [],
        "trailing": [
            SourceComment {
                text: "# trailing",
                position: OwnLine,
                formatted: true,
            },
        ],
    },
}
```

for me (which looks correct). or am i misunderstanding something?

Reply:
Yeah sorry, my mistake. I mixed up the indentation levels. This is correctly handled by the 

```
    if comment_indentation > except_indentation {
        // Delegate to `handle_trailing_body_comment`
        return CommentPlacement::Default(comment);
    }
```

check